### PR TITLE
set domain listener [11472]

### PR DESCRIPTION
**Pull request by IkerLuengo**
**This PR must be merged after #72**

Start reviewing from the commit with description " Rework listener tests assuming set_domain_listener is implemented "

It has some other improvements and bugfixes that were found during this development and/or testing for coverage:
* Statuses must be updated even if no listener is called
* Parameterize the test for listener calls
* Some bugfixes in other tests

### CODE REVIEW

File: test/unittest/StatisticsBackend/CallingUserListenersTests.cpp [hunk 1]
```diff
@@ -311,2562 +369,630 @@ TEST_F(calling_user_listeners_tests, user_discovered)
...
+                    EXPECT_EQ(-1, status.current_count_change);
+                });
+    }
 
-    // Stop the monitor to avoid interfering on next tests
-    StatisticsBackend::stop_monitor(monitor_id);
-}
+    calling_user_listeners_tests_domain_entities()
+    {
+        // Set the profile to ignore discovery data from other processes
```
Comment (JLBuenoLopez-eProsima) [inline test/unittest/StatisticsBackend/CallingUserListenersTests.cpp:769]: Remove repeated line

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
Also, please look at the following lines from the coverage report:

* https://app.codecov.io/gh/eProsima/Fast-DDS-statistics-backend/compare/75/diff#D1L182: a negative test should be done to check this
* https://app.codecov.io/gh/eProsima/Fast-DDS-statistics-backend/compare/75/diff#D2L200
* https://app.codecov.io/gh/eProsima/Fast-DDS-statistics-backend/compare/75/diff#D2L219
* https://app.codecov.io/gh/eProsima/Fast-DDS-statistics-backend/compare/75/diff#D2L237
* https://app.codecov.io/gh/eProsima/Fast-DDS-statistics-backend/compare/75/diff#D2L255
This last ones are similar and I do not understand why the report says that are not covered. This other similar line is on the other hand covered :man_shrugging: https://app.codecov.io/gh/eProsima/Fast-DDS-statistics-backend/compare/75/diff#D2L287

File: test/unittest/StatisticsBackend/StatisticsBackendTests.cpp [hunk 2]
```diff
@@ -325,6 +341,182 @@ TEST_F(statistics_backend_tests, get_data_supported_entity_kinds)
     }
 }
 
+TEST_F(statistics_backend_tests, internal_callbacks_negative_cases)
```
[the remark this answers is not in the document]
Comment (IkerLuengo) [reply test/unittest/StatisticsBackend/StatisticsBackendTests.cpp:344]: I commented through the tests, I believe that should be enough, and so the test can be augmented with other (unrelated) negative cases if necessary in the future

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Comment by JLBuenoLopez-eProsima** [comment]
Windows CI is not building. Please, fix this issue

**Review by JLBuenoLopez-eProsima — DISMISSED** [review]
LGTM

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM.

Windows failure is unrelated.

**Comment by IkerLuengo** [comment]
@Mergifyio rebase

**Comment by MiguelCompany** [comment]
@mergifyio rebase

**Comment by IkerLuengo** [comment]
One testcase is failing:
 statistics_backend_tests.internal_callbacks_negative_cases

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM :+1:

**Pull request merged**